neomodel/exception.py

```diff
@@ -1,6 +1,28 @@
 import importlib
 
 
+class ModelDefinitionMismatch(Exception):
+    """
+    Raised when it is impossible to resolve a Neo4j driver Node to a 
```

Comment:
i had some time getting this, and would propose this alternative:

> Raised when it is impossible to resolve a node from a Neo4j response to a 
defined model. This can either be caused by models that haven't been imported yet, or because
the a returned node has properties that are not defined in the model.

and as noted regarding `relationships.rst` the exception's name seems to describe only the second of its possible causes.

Reply:
on a second thought: there could be two exceptions, one for each case. this should make debugging easier.

Reply:
This was one of the points in discussion with Robin Edwards about what the behaviour should be and having one exception was the current consensus at the time of writing. The condition is "a different set of labels were encountered while trying to retrieve a node". It can arise in different situations. I think that a single exception can describe the problem well enough.